src/events/requests/NewRequestEventHandler.ts

```diff
@@ -1,22 +1,77 @@
-import { Message } from 'discord.js';
+import { Message, TextChannel, RichEmbed } from 'discord.js';
 import * as log4js from 'log4js';
 import EventHandler from '../EventHandler';
+import BotConfig from '../../BotConfig';
+import { ReactionsUtil } from '../../util/ReactionsUtil';
+import MentionCommand from '../../commands/MentionCommand';
 
 export default class NewRequestEventHandler implements EventHandler {
 	public readonly eventName = '';
 
 	private logger = log4js.getLogger( 'NewRequestEventHandler' );
 
+	/**
+	 * A map from request channel IDs to internal channel objects.
+	 */
+	private readonly internalChannels: Map<string, TextChannel>;
+
+	constructor( internalChannels: Map<string, TextChannel> ) {
+		this.internalChannels = internalChannels;
+	}
+
 	// This syntax is used to ensure that `this` refers to the `NewRequestEventHandler` object
-	public onEvent = ( message: Message ): void => {
-		this.logger.info( `User ${ message.author.tag } posted a new request to requests channel ${ message.channel.id }` );
+	public onEvent = async ( origin: Message ): Promise<void> => {
+		if ( origin.type !== 'DEFAULT' ) {
+			if ( origin.deletable ) {
+				origin.delete();
+				return;
+			}
+		}
+
+		this.logger.info( `User ${ origin.author.tag } posted a new request to requests channel ${ origin.channel.id }` );
+
+		await origin.clearReactions();
 
-		if ( message.type === 'PINS_ADD' ) {
-			if ( message.deletable ) {
-				message.delete();
+		const regex = new RegExp( `(?:${MentionCommand.ticketLinkRegex.source}|${MentionCommand.ticketIdRegex.source})`, 'g' );
```

Comment:
Due to the possible restrictions for ticket mentions, the may think you didn't mention any ticket if you escaped the ID or didn't put the required prefix in front of it. In this case, this is not desirable in my opinion. If we decide to require a `!` in front of a ticket ID in order to mention a bot, the bot would not accept requests such as `Please resolve MC-12345 as duplicate of MC-1234`.